model/annotations/task_annotations.go

```diff
@@ -137,3 +137,40 @@ func RemoveSuspectedIssueFromAnnotation(taskId string, execution int, issue Issu
 		bson.M{"$pull": bson.M{SuspectedIssuesKey: issue}},
 	)
 }
+
+func UpdateAnnotation(taskId string, execution int, a *TaskAnnotation, source *Source) error {
+
+	update := bson.M{
+		TaskIdKey:        taskId,
```

Comment:
I'm pretty sure these don't need to be part of the update, since they're part of the query.